src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-storage-class-list/rgw-storage-class-list.component.ts

```diff
@@ -0,0 +1,103 @@
+
+import { RgwZonegroupService } from '~/app/shared/api/rgw-zonegroup.service';
+import { ListWithDetails } from '~/app/shared/classes/list-with-details.class';
+import { StorageClass, CLOUD_TIER , ZoneGroup, Tier_Target, Target } from '../models/rgw-storage-class.model';
+
+@Component({
+  selector: 'cd-rgw-storage-class-list',
+  templateUrl: './rgw-storage-class-list.component.html',
+  styleUrls: ['./rgw-storage-class-list.component.scss']
+})
+export class RgwStorageClassListComponent extends ListWithDetails implements OnInit {
+  columns: CdTableColumn[];
+  selection = new CdTableSelection();
+  tableActions: CdTableAction[];
+  storageClassList: StorageClass[] = [];
+
+  constructor(private rgwZonegroupService: RgwZonegroupService) {
+    super();
+  }
+
+  ngOnInit() {
+    this.columns = [
+      {
+        name: $localize`Zone group`,
+        prop: 'zonegroup_name',
+        flexGrow: 2
+      },
+      {
+        name: $localize`Placement Target`,
+        prop: 'placement_target',
+        flexGrow: 2
+      },
+      {
+        name: $localize`Storage Class`,
+        prop: 'storage_class',
+        flexGrow: 2
+      },
+      {
+        name: $localize`Target Region`,
+        prop: 'region',
+        flexGrow: 2
+      },
+      {
+        name: $localize`Target Endpoint`,
+        prop: 'endpoint',
+        flexGrow: 2
+      },
+    ];
+  }
+
+  loadStorageClass(): Promise<void> {
+    return new Promise((resolve, reject) => {
+      this.rgwZonegroupService.getAllZonegroupsInfo().subscribe(
+        (data: any) => {
+          this.storageClassList = [];
+
+          const tierObj = data.zonegroups
+            .flatMap((zoneGroup: ZoneGroup) =>
+              zoneGroup.placement_targets
+                .filter((target: Target) => target.tier_targets)
+                .flatMap((target: Target) =>
+                  target.tier_targets
+                    .filter((tierTarget: Tier_Target) => tierTarget.val.tier_type === CLOUD_TIER)
+                    .map((tierTarget: Tier_Target) => {
+                      return {
+                        zonegroup_name: zoneGroup.name,
+                        placement_target: target.name,
+                        endpoint: tierTarget.val.s3.endpoint,
+                        access_key: tierTarget.val.s3.access_key,
+                        storage_class: tierTarget.val.storage_class,
+                        target_path: tierTarget.val.s3.target_path,
+                        target_storage_class: tierTarget.val.s3.target_storage_class,
+                        region: tierTarget.val.s3.region,
+                        secret: tierTarget.val.s3.secret,
+                        multipart_min_part_size: tierTarget.val.s3.multipart_min_part_size,
+                        multipart_sync_threshold: tierTarget.val.s3.multipart_sync_threshold,
+                        host_style: tierTarget.val.s3.host_style
+                      };
+                    })
```

Comment:
Lets improve this for readability:

```suggestion
 getTierTargets(tierTarget: Tier_Target, zoneGroupName: string, targetName: string) {
  const { s3 } = tierTarget.val; 
  if (tierTarget.val.tier_type !== CLOUD_TIER) {
    return null;
  }
  return {
    zonegroup_name: zoneGroupName,
    placement_target: targetName,
    endpoint: s3.endpoint,
    access_key: s3.access_key,
    storage_class: s3.storage_class,
    target_path: s3.target_path,
    target_storage_class: s3.target_storage_class,
    region: s3.region,
    secret: s3.secret,
    multipart_min_part_size: s3.multipart_min_part_size,
    multipart_sync_threshold: s3.multipart_sync_threshold,
    host_style: s3.host_style
  };
}
```

And then reuse this.
Also lets try to make less complex the formatting of tier data.
Thanks

Reply:
@afreen23 , I am thinking like as we need all properties from s3 object so can we use spread operator.
Here is the example

  getTierTargets(tierTarget: Tier_Target, zoneGroupName: string, targetName: string) {
    if (tierTarget.val.tier_type !== CLOUD_TIER) return null;
  
    return {
      zonegroup_name: zoneGroupName,
      placement_target: targetName,
      **...tierTarget.val.s3**
    };
  }